src/workerd/api/web-socket.c++

```diff
@@ -633,37 +616,44 @@ size_t countBytesFromMessage(const kj::WebSocket::Message& message) {
 }
 }
 
-kj::Promise<void> WebSocket::pumpAfterFrontOutputLock(
-    IoContext& context, OutgoingMessagesMap& outgoingMessages, kj::WebSocket& ws) {
-  GatedMessage gatedMessage =
-      outgoingMessages.release(*outgoingMessages.ordered().begin());
-  auto size = countBytesFromMessage(gatedMessage.message);
+kj::Promise<void> WebSocket::pump(
+    IoContext& context, OutgoingMessagesMap& outgoingMessages, kj::WebSocket& ws, Native& native) {
+  native.isPumping = true;
+  KJ_DEFER({
+    native.isPumping = false;
```

Comment:
I guess the reason this solves the problem is: Assuming no exceptions, then this defer runs immediately after the `while()` loop below has determined there are no outgoing messages, without yielding to the event loop in between. So it's not possible for some other task to insert a new message that it expects this pump to handle, before `isPumping` becomes `false`.

Maybe worth some comments?